modules/deployment-model-impl/src/main/java/org/mule/runtime/module/deployment/impl/internal/policy/ArtifactExtensionManagerFactory.java

```diff
@@ -77,19 +79,15 @@ public ExtensionManager create(MuleContext muleContext) {
         Map<String, Object> params = new HashMap<>();
         params.put(TYPE_PROPERTY_NAME, extensionManifest.getDescriberManifest().getProperties().get("type"));
         params.put(VERSION, extensionManifest.getVersion());
-
-        extensions.add(new JavaExtensionModelLoader()
-            .loadExtensionModel(artifactPlugin.getArtifactClassLoader().getClassLoader(),
-                                getDefault(extensions),
-                                params));
+        ClassLoader cl = artifactPlugin.getArtifactClassLoader().getClassLoader();
+        extensions.add(new DefaultJavaExtensionModelLoader().loadExtensionModel(cl, getDefault(emptySet()), params));
```

Comment:
It is creating a context where no prior extensions were loaded (the emptySet is the set of extensions in the context). This is OK since the model being loaded does not depend on any other extension.

Adding a `getDefault` in the context that receives no Set doesn't make sense, and overloading the `loadExtensionModel` to create an empty context if none was passed on to it might be a little missleading, since it's important to keep dependencies in mind while loading an extension that is not java-based.